lib/join/server_token.go

```diff
@@ -39,8 +39,11 @@ func (s *Server) handleTokenJoin(
 	// Set any diagnostic info from the ClientParams.
 	setDiagnosticClientParams(stream.Diagnostic(), &tokenInit.ClientParams)
 
-	// There are no additional checks for the token join method, just make the
-	// result message and return it.
+	// make sure the secret provided in the TokenInit matches the token's secret
+	if tokenInit.Secret != token.GetSecret() {
```

Comment:
we should probably use `subtle.ConstantTimeCompare` now that we have that option